packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -124,63 +124,120 @@ export class DataCloudApi {
+                this._constructPartyIdentification(args)
+            )
 
-        const userEngagement = this._concatenateEvents(
+        const contactPointEmail = args.email ? this._concatenateEvents(
             baseEvent,
-            this._generateEventDetails('userEngagement', 'Engagement'),
-            {
-                interactionName: 'page-view',
-                sourceUrl: path
-            }
-        )
+            this._generateEventDetails('contactPointEmail', 'Profile', args.email)
+        ) : null
+
+        return { baseEvent, identityProfile, partyIdentification, contactPointEmail }
+    }
+
+    _handleApiError(err) {
+        if (err?.response?.status === 400) {
+            logger.warn(
+                '[DataCloudApi] 400 Bad Request: Check your Data Cloud configuration (appSourceId, tenantId) and event payload.',
+                {
+                    namespace: 'use-datacloud._handleApiError',
+                    additionalProperties: { error: err?.response }
+                }
+            )
+        } else {
+            logger.error('[DataCloudApi] Error sending Data Cloud event', {
+                namespace: 'use-datacloud._handleApiError',
+                additionalProperties: { error: err?.response }
+            })
+        }
+    }
+
+    /**
+     * Constructs the interaction object and sends it to Data Cloud
+     */
+    _sendInteraction(standardEvents, specificEvents) {
+        const { identityProfile, partyIdentification, contactPointEmail } = standardEvents
 
         const interaction = {
-            events: [...(!this.dnt ? [identityProfile, partyIdentification] : []), userEngagement]
+            events: [
+                ...(!this.dnt ? [identityProfile, partyIdentification] : []),
+                ...(contactPointEmail ? [contactPointEmail] : []),
+                ...specificEvents
+            ]
         }
 
-        this.sdk.webEventsAppSourceIdPost(interaction).catch((err) => this._handleApiError(err))
+        return this.sdk.webEventsAppSourceIdPost(interaction).catch((err) => this._handleApiError(err))
+    }
+
+    /**
+     * Maps search results to DataCloud product format
+     */
+    _mapSearchResultsToProducts(searchResults) {
+        return searchResults?.hits?.map((product) => this._constructDatacloudProduct(product)) || []
+    }
+
+    /**
+     * Sends a `page-view` event to Data Cloud.
+     *
+     * This method records an `userEnagement` event type to track which page the shopper has viewed.
+     *
+     * @param {string} path - The URL path of the page that was viewed
+     * @param {object} args - Additional metadata for the event
+     */
+    async sendViewPage(path, args) {
+        // 1. Create standard events (identity, party, email)
+        const standardEvents = this._createStandardEvents(args, { sourceUrl: path })
+
+        // 2. Create method-specific events using standardEvents.baseEvent
```

Comment:
Nit: we can probably clean up these comments since all they're doing is re-stating what the code is already saying